include/mata/nfa/delta.hh

```diff
@@ -207,37 +316,50 @@ public:
         using pointer = int*;
         using reference = int&;
 
-        explicit const_iterator(const std::vector<Post>& post_p, bool ise = false);
+        explicit transitions_const_iterator(const std::vector<StatePost>& post_p, bool ise = false);
 
-        const_iterator(const std::vector<Post>& post_p, size_t as,
-                       Post::const_iterator pi, StateSet::const_iterator ti, bool ise = false) :
-                post(post_p), current_state(as), post_iterator(pi), targets_position(ti), is_end(ise) {};
+        transitions_const_iterator(const std::vector<StatePost>& post_p, size_t as,
+                                   StatePost::const_iterator pi, StateSet::const_iterator ti, bool ise = false);
 
-        const_iterator(const const_iterator& other) = default;
+        transitions_const_iterator(const transitions_const_iterator& other) = default;
 
-        Trans operator*() const { return Trans{current_state, (*post_iterator).symbol, *targets_position}; }
+        const Trans& operator*() const { return transition; }
 
         // Prefix increment
-        const_iterator& operator++();
+        transitions_const_iterator& operator++();
         // Postfix increment
-        const_iterator operator++(int);
+        const transitions_const_iterator operator++(int);
 
-        const_iterator& operator=(const const_iterator& x);
+        transitions_const_iterator& operator=(const transitions_const_iterator& x);
 
-        friend bool operator==(const const_iterator& a, const const_iterator& b);
-        friend bool operator!=(const const_iterator& a, const const_iterator& b) { return !(a == b); };
+        bool operator==(const transitions_const_iterator& other) const;
+        bool operator!=(const transitions_const_iterator& other) const { return !(*this == other); };
     };
 
-    const_iterator cbegin() const { return const_iterator(posts); }
-    const_iterator cend() const { return const_iterator(posts, true); }
-    const_iterator begin() const { return cbegin(); }
-    const_iterator end() const { return cend(); }
+    transitions_const_iterator transitions_cbegin() const { return transitions_const_iterator(state_posts); }
+    transitions_const_iterator transitions_cend() const { return transitions_const_iterator(state_posts, true); }
+    transitions_const_iterator transitions_begin() const { return transitions_cbegin(); }
+    transitions_const_iterator transitions_end() const { return transitions_cend(); }
 
-private:
-    State find_max_state();
-}; // struct Delta.
+    struct TransitionsIterator {
```

Comment:
just Transitions would be better, the same as Moves

Reply:
Already done with the previous modification of `MovesIterator` to `Moves`.